src/OmniSharp.LanguageServerProtocol/Handlers/CodeLensHandler.cs

```diff
@@ -0,0 +1,128 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using OmniSharp.Extensions.JsonRpc;
+using OmniSharp.Extensions.LanguageServer.Capabilities.Client;
+using OmniSharp.Extensions.LanguageServer.Models;
+using OmniSharp.Extensions.LanguageServer.Protocol;
+using OmniSharp.Models;
+using OmniSharp.Models.FindUsages;
+using OmniSharp.Models.MembersTree;
+
+namespace OmniSharp.LanguageServerProtocol.Handlers
+{
+    internal sealed class CodeLensHandler : ICodeLensHandler, ICodeLensResolveHandler
+    {
+        public static IEnumerable<IJsonRpcHandler> Enumerate(RequestHandlers handlers)
+        {
+            foreach (var (selector, membersAsTreeHandler, findUsagesHandler) in handlers
+                .OfType<
+                    Mef.IRequestHandler<MembersTreeRequest, FileMemberTree>,
+                    Mef.IRequestHandler<FindUsagesRequest, QuickFixResponse>>())
```

Comment:
After adding this (in second commit) this method will now return null. Haven't had time to debug yet. Without the resolver functionalty, the code lens handler works ok. @david-driscoll do you spot any errors which could lead to this? The same handler is used in #1070